lib/puppet/functions/split.rb

```diff
@@ -21,29 +21,32 @@
 #
 Puppet::Functions.create_function(:split) do
   dispatch :split_String do
-    param 'String', :str
+    param 'Variant[String, Sensitive[String]]', :str
     param 'String', :pattern
   end
 
   dispatch :split_Regexp do
-    param 'String', :str
+    param 'Variant[String, Sensitive[String]]', :str
     param 'Regexp', :pattern
   end
 
   dispatch :split_RegexpType do
-    param 'String', :str
+    param 'Variant[String, Sensitive[String]]', :str
     param 'Type[Regexp]', :pattern
   end
 
   def split_String(str, pattern)
+    str = str.unwrap if str.respond_to?(:unwrap)
```

Comment:
Hi @cocker-cc, thank you for your contribution! This looks good, but I think it'd be cleaner to define different dispatchers for the `Sensitive[String]` cases. That way we don't have to later "guess" whether `str` is sensitive or not, based on `respond_to`:

```ruby
dispatch :split_String do
  param 'String', :str
  param 'String', :pattern
end

dispatch :split_Sensitive_String do
  param 'Sensitive[String]', :sensitive
  param 'String', :pattern
end

def split_String(str, pattern)
  str.split(Regexp.compile(pattern))
end

def split_Sensitive_String(sensitive, pattern)
  split_String(sensitive.unwrap, pattern)
end
```

Reply:
ACK. I amended my new Commit to the PR.